modules/030-cloud-provider-aws/images/cloud-data-discoverer/discoverer.go

```diff
@@ -114,3 +114,7 @@ func (d *Discoverer) InstanceTypes(_ context.Context) ([]v1alpha1.InstanceType,
 
 	return res, nil
 }
+
+func (d *Discoverer) DiscoveryData(ctx context.Context) ([]byte, error) {
+	return []byte("{}"), nil
```

Comment:
lets return nil and if function return nil without error will not create secret with data.

Reply:
no secret is better than a knowingly invalid secret